src/dpg_slam/dpg_slam.cc

```diff
@@ -613,33 +602,106 @@ namespace dpg_slam {
+	std::unordered_set<cellKey, boost::hash<cellKey>> currentUncoveredCells;
+	bool isLocalSubmapInitialized = false;
+
+	// populate the list of uncovered cells
+	for (uint32_t i=0; i<poseChainGrids.size(); i++) {
+		std::unordered_map<cellKey, CellStatus, boost::hash<cellKey>> currNodeGridInfo = poseChainGrids[i].getGridInfo();
+		for (const auto &kv : currNodeGridInfo) { 
+			currentUncoveredCells.insert(kv.first);
+		}
+	}
+	uint64_t totalUncoveredCells = currentUncoveredCells.size();
+	uint64_t currentUncoveredCellsSize  = currentUncoveredCells.size();
+	
+	// compute submap that covers the currentUncoveredCells
+	// Get max laser range
+	float proximityThreshold = dpg_parameters_.distance_threshold_for_local_submap_nodes_;
+	float coverageThreshold = dpg_parameters_.current_pose_graph_coverage_threshold_;
+
+	for (uint32_t j=0; j<(dpg_nodes_.size()-current_pass_nodes_.size()); j++) { 
+		
+		DpgNode pastNode = dpg_nodes_[j];
+		if (!pastNode.isNodeActive())
+			continue;
+
+                std::pair<Vector2f, float> pastNodeInfo = pastNode.getEstimatedPosition();
+                Vector2f pastNodePosition = pastNodeInfo.first;
+		// checking if the past node is close to any of the current nodes.
+		bool isInProximity = false;
+		for (uint32_t i=0; i<poseChain.size(); i++) {
+			DpgNode currNode = poseChain[i];
+                	std::pair<Vector2f, float> currentNodeInfo = currNode.getEstimatedPosition();
+               		Vector2f currentNodePosition = currentNodeInfo.first;
+			float distanceToPastNode = (currentNodePosition - pastNodePosition).norm();
+			if (distanceToPastNode <= proximityThreshold)
+				break;
+		}		
+		
+		if (!isInProximity)
+			continue;	
+
+		if (!isLocalSubmapInitialized) {
+			occupancyGrid tempCombinedGrid(dpg_nodes_[j], dpg_parameters_, pose_graph_parameters_);
+			getUpdatedCoverageForCurrentPoseChain(currentUncoveredCells, tempCombinedGrid);
+			uint32_t newCurrentUncoveredCellsSize  = currentUncoveredCells.size();
+			if (newCurrentUncoveredCellsSize < currentUncoveredCellsSize) {
+                        	subMapOccupancyGrid = tempCombinedGrid;
+                        	isLocalSubmapInitialized = true;
+                        	currentUncoveredCellsSize = newCurrentUncoveredCellsSize;
+                	}
+		} else{
+			occupancyGrid tempGrid(dpg_nodes_[j], dpg_parameters_, pose_graph_parameters_);
+                        occupancyGrid tempCombinedGrid(subMapOccupancyGrid, tempGrid, dpg_parameters_, pose_graph_parameters_);
+			getUpdatedCoverageForCurrentPoseChain(currentUncoveredCells, tempCombinedGrid);
+			uint32_t newCurrentUncoveredCellsSize  = currentUncoveredCells.size();
+			if (newCurrentUncoveredCellsSize < currentUncoveredCellsSize) {
+				subMapOccupancyGrid = tempCombinedGrid;
+				currentUncoveredCellsSize = newCurrentUncoveredCellsSize;
+			}
+		}
+
+		double currentCoverage = 1 - (double)currentUncoveredCellsSize/totalUncoveredCells;
```

Comment:
I'm not positive, but. I think you'll need to put the casted variable in parenthesis with its quotes. 
``` 1 - ((double) currentUncoveredCellsSize) / totalUncoveredCells;```
I think this will just cast the already truncated result to a double which is not what we want.